pytorch_lightning/strategies/launchers/tpu_spawn.py

```diff
@@ -0,0 +1,106 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import time
+from multiprocessing.queues import SimpleQueue
+from typing import Any, Callable, Optional
+
+import torch.multiprocessing as mp
+
+import pytorch_lightning as pl
+from pytorch_lightning.strategies.launchers.ddp_spawn import _FakeQueue, _SpawnOutput, DDPSpawnLauncher
+from pytorch_lightning.strategies.strategy import Strategy
+from pytorch_lightning.trainer.states import TrainerFn
+from pytorch_lightning.utilities import _TPU_AVAILABLE
+from pytorch_lightning.utilities.apply_func import move_data_to_device
+from pytorch_lightning.utilities.distributed import rank_zero_debug
+from pytorch_lightning.utilities.model_helpers import is_overridden
+
+if _TPU_AVAILABLE:
+    import torch_xla.distributed.xla_multiprocessing as xmp
+else:
+    xm, xmp, MpDeviceLoader, rendezvous = [None] * 4
+
+
+class TPUSpawnLauncher(DDPSpawnLauncher):
```

Comment:
let's avoid inheritance wherever possible

Reply:
but there are some methods from spawn launcher required here. Else it would need to replicate those here.